WcaOnRails/app/models/registration_competition_event.rb

```diff
@@ -7,4 +7,10 @@ class RegistrationCompetitionEvent < ApplicationRecord
   validates :competition_event, presence: true
 
   has_one :event, through: :competition_event
+
+  def event
+    # This may look silly, but actually CompetitionEvent is using a cache
+    # that would normally be circumvented by the default has_one_through association
+    competition_event&.event
```

Comment:
Naive question: have you tried `delegate :event, to: :competition_event` instead of implementing the method yourself?
I actually don't know what happens if `competition_event` is nil but maybe it "just works".

Reply:
Thanks for the idea! Turns out there is `allow_nil: true` for precisely such a case.

Reply:
Awesome!